File: LeetCode-Problems/Easy/Kid-With-Most-Candies.py
# Python
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """
        booleanList = list()
        max = 0;
        for x in candies:
            if x > max:
                max = x  
        for x in candies:
            if x + extraCandies >= max:
                booleanList.append(True)
            else:
                booleanList.append(False)
        return booleanList

File: LeetCode-Problems/Easy/Kid-With-Most-Candies.java
//Java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        ArrayList<Boolean> returnedList = new ArrayList<Boolean>(candies.length);
        int max = 0;
        for (int i = 0; i < candies.length; i++){
            if (candies[i] > max){
                max = candies[i];
            }
        }
        for (int i = 0; i < candies.length; i++){
            if (candies[i] + extraCandies >= max){
                returnedList.add(true);
            }
            else{
                returnedList.add(false);
            }
        }
        return returnedList;
    }
}
